x/ccv/provider/types/codec.go

```diff
@@ -36,6 +36,11 @@ func RegisterInterfaces(registry codectypes.InterfaceRegistry) {
 		&EquivocationProposal{},
 	)
 
+	registry.RegisterImplementations(
+		(*sdk.Msg)(nil),
```

Comment:
Pattern matching: Should the MsgSubmitConsumerDoubleVoting also have something like this?

Reply:
Good catch